src/main/webapp/js/adminEmailLog.js

```diff
@@ -65,27 +50,25 @@ function submitFormAjax(offset) {
             button.html('Retry');
         },
         success: function(data) {
-            setTimeout(function() {
-                if (data.isError) {
-                    setFormErrorMessage(button, data.errorMessage);
-                } else {
-                    // Inject new log row
-                    var logs = data.logs;
-                    $.each(logs, function(i, value) {
-                        lastLogRow.after(value.logInfoAsHtml);
-                        lastLogRow = $('#emailLogsTable tr:last');
-                        bindClickAction();
-                        clickOlderButtonIfNeeded();
-                    });
-                }
-
-                setStatusMessage(data.statusForAjax, StatusType.INFO);
-
-            }, 500);
+            var $data = $(data);
+            $logsTable.append($data.find('#logs-table > tbody').html());
+            bindClickAction();
+            highlightKeywordsInEmailLogMessage();
+            setStatusMessage($data.find('#status-message').html(), StatusType.INFO);
         }
     });
 }
 
 function setFormErrorMessage(button, msg) {
     button.after('&nbsp;&nbsp;&nbsp;' + msg);
 }
+
+/**
+ * Highlights default/search keywords in eamil log message.
+ */
+function highlightKeywordsInEmailLogMessage() {
+    $('.email-receiver').highlight($('#query-keywords-receiver').val().split(','));
+    $('.email-subject').highlight($('#query-keywords-subject').val().split(','));
+    $('.email-content').highlight($('#query-keywords-content').val().split(','));
```

Comment:
The invisible content can be trigger by clicking.

![demo 1](https://cloud.githubusercontent.com/assets/13977235/23692093/9548002e-0407-11e7-9351-a65e936bc3aa.gif)